[slidings] iP

slidings opened the pull request:
## Pebble chat bot features 🪨 🪨 
Task management from CLI

# Features
- Types of tasks includes ToDos, Deadlines, Events
- Includes CRUD functionality
- Includes filtering list by description

# How to use?
1. Fork this [Repo](https://github.com/slidings/ip)
2. Install on your local device
>[!TIP]
>Use git clone git@github.com:<username>/ip.git into the folder you want to run this code on
3. Run main function in ```Pebble.java```
```java
    public static void main(String[] args) {
        // launch with data file
        new Pebble("data/pebble.txt").run();
    }
```

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

__Code of the day__
> _"To be is to do."_ - Socrates
> _"To do is to be."_ - Kant
> _"Do be do be do."_ - Scooby-Doo

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

slidings committed:
added .class to .gitignore

slidings committed:
renamed chatbot

slidings committed:
Greet and exit

slidings committed:
Echoing chatbot

slidings committed:
Basic list

slidings committed:
Markable list

slidings committed:
Added ToDo Event and Deadline

slidings committed:
added trivial test cases

slidings committed:
added exceptions

slidings committed:
delete feature

slidings committed:
refactor code into enum

slidings committed:
Add feature 1 branch

slidings committed:
Add masterBranch file

slidings committed:
Merge branch 'master' into feature1

Merging test for feature 1 onto branch

slidings committed:
Update remote

slidings committed:
Edit feature1 (Test)

slidings committed:
Edit masterbranch (Test)

slidings committed:
Sync feature1 branch with master branch

slidings committed:
Update (Test)

slidings committed:
Update (Test)

slidings committed:
Merge branch 'master' into feature1 part 2

slidings committed:
Reset to master branch

slidings committed:
Merge branch 'master' into feature1

slidings committed:
Remove backup files

slidings committed:
Add commit feature1

slidings committed:
Add commit master branch

slidings committed:
Merge branch 'feature1'

slidings committed:
Add Singapore

slidings committed:
Edit Singapura

slidings committed:
Merge branch 'add-countries'

slidings committed:
Add level 7 branch

slidings committed:
Remove unrelated files from iP

slidings committed:
Merge branch 'Level-7'

slidings committed:
Re-add branch-Level-7 (Fix previous mistake)

slidings committed:
Fix previous branch mistake, merge branch 'branch-Level-7'

slidings committed:
Add Level 8 Date conversion

slidings committed:
Add Level 8 Date conversion

slidings committed:
Merge branch 'branch-Level-8'

slidings committed:
Refactor for better OOP

slidings committed:
Merge branch 'branch-A-MoreOOP'

slidings committed:
Add all files into package

slidings committed:
Merge branch 'add-gradle-support'

slidings committed:
Configure for gradle

slidings committed:
Add JUnit test case for Deadline and Event

slidings committed:
Configure fat JAR

slidings committed:
Add find description feature

slidings committed:
Merge branch 'branch-Level-9'

slidings committed:
Merge branch-Level-9

slidings committed:
Add javaDoc comments

slidings committed:
Merge branch 'branch-A-JavaDoc'

slidings committed:
Fix Coding Standard

slidings committed:
Merge branch 'branch-A-CodingStandard'

Slipperybara commented on src/main/java/Pebble/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+    protected String stringBy;
+
+    /**
+     *  Constructor for Deadline.
```
Comments for methods should start with verbs, not sure if its the case for constructors?

Slipperybara commented on src/main/java/Pebble/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ * Subclass of Tasks that has a deadline.
+ */
+public class Deadline extends Task {
+    /** When deadline date is properly formatted */
```
I like how well documented your code is 👍

Slipperybara commented on src/main/java/Pebble/InvalidTaskNumberException.java:
```diff
@@ -0,0 +1,8 @@
+
+// Custom exception for unknown commands (unused)
+public class InvalidTaskNumberException extends Exception {
+        public InvalidTaskNumberException() {
```
Here the indentation should be 4 spaces?

slidings replied:
oops must have double indented, thanks for flagging it!

Slipperybara commented on src/main/java/Pebble/Parser.java:
```diff
@@ -0,0 +1,99 @@
+     * @param line Line read from txt file.
+     * @return Instance of task corresponding to the saved line.
+     */
+    public static Task parseTaskFromString(String line) {
```
I like your naming of methods, very clear 💯

Slipperybara commented on src/main/java/Pebble/Pebble.java:
```diff
@@ -0,0 +1,60 @@
+    private TasksList tasksList;
+
+    /**
+     * Initialises the whole program with the data file.
```
I like your documenting details

Slipperybara reviewed (commented):
LGTM, only one line of code have spacing change

DiegoTheExplorar commented on src/main/java/Pebble/Parser.java:
```diff
@@ -0,0 +1,99 @@
+package pebble;
```
Good job on this. I did not know packages had to be in lowercase. All my packages started with a capital :{

DiegoTheExplorar commented on src/main/java/Pebble/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        String command = words[0];
+        String arguments = words.length > 1 ? words[1] : "";
+
+        switch (command) {
+        case "bye":
+            return new Command(CommandType.BYE, arguments);
+        case "list":
+            return new Command(CommandType.LIST, arguments);
+        case "mark":
+            return new Command(CommandType.MARK, arguments);
+        case "unmark":
+            return new Command(CommandType.UNMARK, arguments);
+        case "todo":
+            return new Command(CommandType.TODO, arguments);
+        case "deadline":
+            return new Command(CommandType.DEADLINE, arguments);
+        case "event":
+            return new Command(CommandType.EVENT, arguments);
+        case "delete":
+            return new Command(CommandType.DELETE, arguments);
+        case "find":
+            return new Command(CommandType.FIND, arguments);
+        default:
+            return new Command(CommandType.UNKNOWN, arguments);
+        }
```
This looks very neat. Nice job on using switch case (unlike me I just used if else like a goon :{  )

DiegoTheExplorar commented on src/main/java/Pebble/Storage.java:
```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+<<<<<<< Updated upstream
+ * 
+=======
+ * Class to handle storage of tasks list on local device.
+>>>>>>> Stashed changes
+ */
```
Do take not of this. I believe there was some issue with you stashing changes and not committing them but I am not too sure.

slidings replied:
oh i have no idea that was there, thanks for flagging this to me!! 🙏

DiegoTheExplorar commented on src/main/java/Pebble/Ui.java:
```diff
@@ -0,0 +1,82 @@
+        showLine();
+    }
+
+    public void showInvalidTaskNumber() {
+        showLine();
+        System.out.println("    Invalid task number.");
+        showLine();
+    }
```
I am just curious, why did you leave a few spaces before the message you print?

slidings replied:
Initially i tried to output exactly the same way as Duke did haha, thinking that there were tests cases

DiegoTheExplorar reviewed (commented):
Overall nice job :}

slidings committed:
Fixing changes from code review

slidings committed:
Add checkstyle and fix styling errors

slidings committed:
Refactor Pebble into GUI

slidings committed:
Refactor Pebble into GUI

slidings committed:
Merge branch 'branch-Level-10'

slidings committed:
Add git message and explain Level-10

The previous version of Pebble was interacted through the Command Line
Interface.

This is not very user friendly as it requires users to know the basics
of running the code in a code editor.

A Graphic User Interface (GUI) is added to provide users another way
of interacting with Pebble, using JavaFX, FXML and CSS. This way a
fatJar file can be created to encapsulate the code into an executable,
making it even more user friendly.

slidings committed:
Add assertions

slidings committed:
Refactor Parser.java to adhere to SLAP

slidings committed:
Refactor command.java

slidings committed:
Merge pull request #2 from slidings/branch-A-CodeQuality

Refactor Parser.java to adhere to SLAP

slidings committed:
Merge pull request #1 from slidings/branch-A-Assertions

Add assertions

slidings committed:
Add friendlier syntax options

slidings committed:
Merge branch 'branch-C-FriendlierSyntax'

slidings committed:
Fix show list bug

slidings committed:
Remove magic numbers and add javadoc

slidings committed:
Add user guide

slidings committed:
Improve formatting for README.md

slidings committed:
Add javadocs and edit exception handling

First, there is missing documentation in many public methods. Second
exceptions are not handled properly with trivial or redundant
exceptions.

Adding javadocs allows for better documentation for other people as well
as for future self. Better exception handling allows for less convoluted
code and easier debugging in the future.

Javadocs is added to most public methods and fields. Exceptions are
changed to a tigher scope. Invalid task class is converted into
InvalidTaskException.

Invalid task class is removed to completely prevent users from
accidentally creating a new redundant class.

slidings committed:
Add assertion

slidings committed:
Add Junit test for Storage class

slidings committed:
Format README.md file

slidings committed:
Format README.md file

slidings committed:
Edit README.md formatting
